[DataGrid] Remove showToolbar prop (#948)
mui-org/material-ui-x: docs/src/pages/components/data-grid/rendering/rendering.md

### Density selector

> To enable the density selector you need to add the `Header: GridHeader` the `components` grid prop.

Reviewer: ```suggestion
To enable the density selector you need to add the `Header: GridHeader` to the `components` grid prop.
```

### Density selector

> To enable the density selector you need to compose your own toolbar using the `DensitySelector` and provide it to the `Toolbar` in the `components` grid prop.

Reviewer: ```suggestion
To enable the density selector you need to compose a toolbar containing the `DensitySelector` component, and apply it using the `Toolbar` key in the grid `components` prop.
```

### Toolbar

To enable the toolbar you need to add the `Toolbar: GridHeader` to the `components` grid prop.
This demo showcases how this can be achieve.

{{"demo": "pages/components/data-grid/rendering/ToolbarGrid.js", "bg": "inline"}}

> Alternatively you can compose your own toolbar.

Reviewer 2: I think that we should avoid the need to open the source:

````suggestion
Alternatively, you can compose your own toolbar.

```jsx
function CustomToolbar() {
  return (
    <GridToolbarContainer>
      <ColumnsToolbarButton />
      <FilterToolbarButton />
    </GridToolbarContainer>
  );
}
```
````

### Toolbar

> To enable the toolbar you need to add the `Toolbar: GridHeader` to the `components` grid prop.

Reviewer: ```suggestion
To enable the toolbar you need to add the `Toolbar: GridToolbar` to the grid `components` prop.
```
